diagrams/lib/tile_mode.dart

```diff
@@ -129,8 +129,11 @@ class TileModeDiagramStep extends DiagramStep {
 
     final List<File> outputFiles = <File>[];
     for (TileModeDiagram tileModeDiagram in tileModeDiagrams) {
+      if (onlyGenerate != null && !onlyGenerate.contains(tileModeDiagram.name)) {
+        continue;
+      }
```

Comment:
it's a pity we have to duplicate this pattern everywhere. not sure how we can avoid it... maybe just filter by `category` instead, rather than by any specific filename? That way you could factor the check out into the superclass...

Reply:
There already is the ability to filter by category (see `--category` on generate.dart). This just refines it (or provides a separate filter) for filenames.  I thought about making a separate base class that represents a single diagram and has a name (maybe Diagram?).  Each DiagramStep can then provide a list of Diagrams from an accessor, and the base generateDiagrams would call generateDiagram for each one where the filename matches, instead of subclasses overriding generateDiagrams.  I'll see if that works well.

Reply:
OK, I refactored it so that the filtering happens in the step.  It does seem to make things a bit cleaner for the implementations.